developer/contributing/pull-requests.rst

```diff
@@ -9,15 +9,37 @@ When creating a pull request:
   example: ``feature/what-my-pr-does``. Note that dashes should be used
   instead of spaces (not underscores).
 * Add your feature/bugfix/enhancement.
+* Create atomic and logically separate commits (use the power of ``git rebase``
+  to have a clean and logical history).
+* Write good commit messages (see the tip below).
 * Write tests.
 * :doc:`test-your-code`.
 * Add a line in the format of "<type> <pr-number>
   [<affected bundle or component>] <description>" to the ``CHANGELOG.md`` file
   in the root directory.
-* In case you have some changes breaking backwards compability you also have to
+* In case you do changes that break backwards compatibility you also have to
   add a description to the ``UPGRADE.md`` file in the root directory.
-* Use a meaningful name for the pull request.
+* Use a meaningful name for the pull request (see the tip below).
 * Create the pull request as soon as possible.
+* In addition to a pull request to the code repository, you should also send a
+  pull request to the `documentation repository`_ to update the documentation when appropriate.
+
+.. tip::
+
+    A good commit message is composed of a summary (the first line),
+    optionally followed by a blank line and a more detailed description. The
+    summary should start with the Component you are working on in square
+    brackets (``[MediaBundle]``, ``[ContentBundle]``, ...). Use a
+    verb (``Fixed ...``, ``Added ...``, ...) to start the summary and don't
+    add a period at the end.
+
+.. tip::
+
+    A good pull request title should start with the Component you are working on
+    in square brackets (``[MediaBundle]``, ``[ContentBundle]``, ...). Use a
+    verb (``Fixed ...``, ``Added ...``, ...) to start the title and don't
+    add a period at the end. If your pull request is not finished yet prefix the
+    title with ``[WIP]``.
```

Comment:
Prepending ` [WIP]` is also not happening internally because we use [ZenHub](https://www.zenhub.io/) for that. How should we handle this? I don't think there is an easy way to open ZenHub for external users.